docs/source/sci_api_tutorial/armer.rst

```diff
@@ -2,21 +2,280 @@
-
-TODO
-  
+
+However, none of the function will be directly access by user in normal EnzyHTP workflows. They are used by developers in APIs of EnzyHTP. But all of the APIs that uses ARMer will expose an input that allow users to config the resource.
+
+.. dropdown:: :fa:`eye,mr-1` Developer Integration with Science APIs`
+
+    Developers using EnzyHTP can directly leverage ARMer's capabilities through the platform's APIs. 
+    
+    In EnzyHTP, ARMer is integrated into the workflow, which includes sub-tasks like mutant generation, molecular dynamics simulations, quantum mechanical calculations, and data analysis. 
+    Each subtask was a seperate science API that has different computing needs, which ARMer manages effectively.
+
+    Developers can utilize ARMer to tailor the computational resources specifically for the task at hand, whether it involves intensive CPU usage for molecular dynamics simulations or GPU resources for more complex quantum mechanical calculations.
+
+.. dropdown:: :fa:`eye,mr-1` User Interaction via Configured API`
+
+    For users, the interaction with ARMer is streamlined through configurations exposed by API developers:
+
+    Cluster Job Configuration Dictionary: This dictionary (cluster_job_config) shown above is provided by the API developers and exposes various configurable options that users can set according to their specific requirements. It includes parameters such as cluster type, environmental settings, and resource keywords.
+    
+    Simplified Job Submission: Users don’t need to manage complex cluster configurations directly. Instead, they provide necessary parameters through a high-level interface, simplifying the computational aspects of enzyme modeling.
+
+
+Specifically, only the `Job configuration` requires specific input from different users.
+
+The first step involves configuring a job using the config_job method. This method prepares the submission script by specifying the commands to be executed, environmental settings, and resource requirements.
+
+.. admonition:: Here is the `cluster_job_config` dictionary
+        
+        These parameters are specified under the argument ``res_keywords``.
+
+        ``core_type``: This specifies that the job should be run on GPU/CPU cores. 
+
+        ``nodes``: How many nodes needed to request for the job.
+
+        ``node_cores``: How many cores needed for each node. If GPU is used, usually only one core per node will be requested.
+
+        ``job_name``: This sets the name of the job to "job_name". You can change this to a more descriptive name for your job.
+
+        ``partition``: This specifies that the job should be submitted to a specific partition, which is likely a partition dedicated to GPU resources.
+        
+        ``mem_per_core``: This requests a number of gigabytes of memory per core.
+        
+        ``walltime``: This sets the maximum walltime (execution time) for the job. '24:00:00' means 24 hours.
+        
```

Comment:
rearrange this section to align with the hierarchy of the actual dictionary.

Reply:
also be aware of the confusion Yinjie made. emphasize these keywords are build-in keywords in EnzyHTP. it is not the same as the keywords of SLURM or other job scheduling system

Reply:
By the "hierarchy of the actual dictionary" I mean like this:
```
There are 2 key-value pairs accepted in this dictionary, ``cluster`` and ``res_keywords``.
``cluster``
         The value under this key defines ... as a ClusterInterface() object
         Example: ...
``res_keywords``
         The value under this key defines ... as a dictionary
         <more to say about what k-v pairs of this dictionary>
         Example: ...
```

Reply:
Also we need to explain the "default" behavior of the dictionary under `res_keywords` when a key is not specified.